glib/src/collections/slist.rs

```diff
@@ -652,7 +652,7 @@ pub struct IterMut<'a, T: TransparentPtrType> {
 
 impl<'a, T: TransparentPtrType> IterMut<'a, T> {
     #[inline]
-    fn new(list: &'a mut SList<T>) -> IterMut<'a, T> {
```

Comment:
This is wrong. The iterator gives mutable access to the list so we need a mutable reference